src/lib/stream.rs

```diff
@@ -567,22 +542,12 @@ impl<'a> CodedInputStream<'a> {
         self.read_unknown(wire_type).map(|_| ())
     }
 
-    /// Read raw bytes into supplied vector. Vector must be empty.
+    /// Read raw bytes into the supplied vector.  The vector will be resized as needed and
+    /// overwritten.
     pub fn read_raw_bytes_into(&mut self, count: u32, target: &mut Vec<u8>) -> ProtobufResult<()> {
-        assert!(target.is_empty());
         target.reserve(count as usize);
```

Comment:
This doesn't work properly now if `target` is not empty, because `reserve` reserves _additional_ bytes. You need to either return the assertion or `.set_len(0)` first.